crates/autopilot/src/solvable_orders.rs

```diff
@@ -604,39 +611,39 @@ async fn update_task(
     }
 }
 
-fn get_orders_with_native_prices(
+async fn get_orders_with_native_prices(
     orders: Vec<Order>,
     native_price_estimator: &CachingNativePriceEstimator,
     metrics: &Metrics,
+    additional_tokens: Vec<H160>,
+    timeout: Duration,
 ) -> (Vec<Order>, BTreeMap<H160, U256>) {
     let traded_tokens = orders
         .iter()
         .flat_map(|order| [order.data.sell_token, order.data.buy_token])
-        .collect::<HashSet<_>>()
-        .into_iter()
-        .collect::<Vec<_>>();
+        .chain(additional_tokens.into_iter())
+        .collect::<HashSet<_>>();
 
-    let prices = get_native_prices(&traded_tokens, native_price_estimator);
+    let prices = get_native_prices(
+        &traded_tokens.into_iter().collect::<Vec<_>>(),
+        native_price_estimator,
+        timeout,
+    )
+    .await;
 
     // Filter both orders and prices so that we only return orders that have prices
     // and prices that have orders.
```

Comment:
This comment is no longer correct. Now we can return that prices that have no orders.